docs/environments/developer.yml

```diff
@@ -25,13 +25,13 @@ dependencies:
   - pooch
   - pre-commit
   - pytest
-  - python=3.10
+  - python=3.12
```

Comment:
Why? We should develop with the minimal supported version to avoid accidentally using newer features.

Reply:
Hmm, I would personally prefer the CI catch these things but I won't fight it :)

Reply:
That is later than catching it locally. So it's more work. Plus, when you lock dependencies, you have to use the same Python version locally.

Reply:
ah yeah true!